File: sdkwork-backend-sdk-typescript/src/api/ai-audio-effect-generation.ts
import { backendApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { GenerateAudioEffectForm, PlusApiResultGenerateAudioEffectVO } from '../types';


export class AiAudioEffectGenerationApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Create audio effect generation task */
  async create(body: GenerateAudioEffectForm): Promise<PlusApiResultGenerateAudioEffectVO> {
    return this.client.post<PlusApiResultGenerateAudioEffectVO>(backendApiPath(`/generation/audio-effect/create`), body);
  }

/** Get audio effect generation result */
  async getResult(taskId: string | number): Promise<PlusApiResultGenerateAudioEffectVO> {
    return this.client.get<PlusApiResultGenerateAudioEffectVO>(backendApiPath(`/generation/audio-effect/result/${taskId}`));
  }
}

export function createAiAudioEffectGenerationApi(client: HttpClient): AiAudioEffectGenerationApi {
  return new AiAudioEffectGenerationApi(client);
}

File: sdkwork-backend-sdk-go/api/ai_audio_effect_generation.go
package api

import (
    "fmt"
    sdktypes "github.com/sdkwork/backend-sdk/types"
    sdkhttp "github.com/sdkwork/backend-sdk/http"
)

type AiAudioEffectGenerationApi struct {
    client *sdkhttp.Client
}

func NewAiAudioEffectGenerationApi(client *sdkhttp.Client) *AiAudioEffectGenerationApi {
    return &AiAudioEffectGenerationApi{client: client}
}

// Create audio effect generation task
func (a *AiAudioEffectGenerationApi) Create(body sdktypes.GenerateAudioEffectForm) (sdktypes.PlusApiResultGenerateAudioEffectVO, error) {
    raw, err := a.client.Post(BackendApiPath("/generation/audio-effect/create"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultGenerateAudioEffectVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultGenerateAudioEffectVO](raw)
}

// Get audio effect generation result
func (a *AiAudioEffectGenerationApi) GetResult(taskId string) (sdktypes.PlusApiResultGenerateAudioEffectVO, error) {
    raw, err := a.client.Get(BackendApiPath(fmt.Sprintf("/generation/audio-effect/result/%s", taskId)), nil, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultGenerateAudioEffectVO
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultGenerateAudioEffectVO](raw)
}
